sdk/python/kfp/compiler/_op_to_template.py

```diff
@@ -176,9 +176,13 @@ def _op_to_template(op: dsl.ContainerOp):
     processed_op = _process_container_ops(op)
 
     # default output artifacts
+    output_artifact_paths = {}
+    output_artifact_paths.setdefault('mlpipeline-ui-metadata', '/mlpipeline-ui-metadata.json')
```

Comment:
why use the setdefault function?

Reply:
Right not it makes not difference.
But in future, when we change `output_artifact_paths = {}` to something non-empty, the `.setdefault` call will set the paths for the 'mlpipeline-ui-metadata' artifact only if it wasn't set before.

We have some customers (IBM) who explicitly want to customize those metadata file paths.

Reply:
Do not think it is necessary now.
But will approve since it is very minor point.

Reply:
You're probably right. This looks a bit strange at this state. Hope it's not for long.

My train of thoughts was:
I saw that PR from IBM (#998) became conflicted after #879 was merged (due to this part being moved). I had to restore #900, but decided to restore it in a way it's done in #998 to reduce the later diff.
https://github.com/kubeflow/pipelines/pull/998/files#diff-618bb1fbca23caf0b3edc26f1b38caa8